quattor/types/component.pan

```diff
@@ -18,6 +18,6 @@ type structure_component = extensible {
     "version"           ? string with match(SELF, '^\d+\.\d+\.\d+$')
     "register_change"   ? element*[]
     "dependencies"      ? structure_component_dependency
-    "code"              ? structure_component_code
+    "code"              ? structure_component_code with deprecated(0, "Shipping component code in the profiles is deprecated and is non-functional.")
```

Comment:
should return `true` like 
https://github.com/quattor/configuration-modules-core/blob/master/ncm-metaconfig/src/main/pan/components/metaconfig/schema.pan#L18